util/testutil/util.go

```diff
@@ -181,3 +186,12 @@ func RandomDigest() string {
 	d := digest.FromBytes(RandomByteData(10))
 	return d.String()
 }
+
+// RandString returns a random string of length n
+func RandString(n int) string {
```

Comment:
File names must be random since we use them to traverse the fs tree when creating the DB and so RandString does not use our seeded random. We still use a fixed size, so there shouldn't be any variance between runs.

Reply:
I don't understand this statement.

Reply:
If we used the seeded random all our filenames would be the same and the only thing differentiating them would be depth level (eg: `file` vs `file/file` vs `file/file/file`). This is OK when writing to our nodes bucket since we don't care about relationships between nodes. This doesn't hold true when writing to our metadata bucket, though.

Every node is identified by some ID and has a corresponding metadata entry. The metadata entry has a map of it's children with the child name as the key. When we are adding children to the metadata entry of a node we will end up overwriting any existing children if they share the same name, which never happens in practice since you cannot multiple children nodes/files with the same name under a single parent node/directory. What this means in practice, is that our metadata bucket will not be fully populated.

Instead, we use rand so we can get an actual pseudo random string. We still use a fixed length of 10 for the filename, to avoid any variance between runs.

Reply:
If we used the seeded random all our filenames would be the same and the only thing differentiating them would be depth level (eg: `file` vs `file/file` vs `file/file/file`).

When looping through the TOC we maintain a map of node ID to metadata entry. The metadata entry has a map of the nodes children where the child name is the key. When we are adding children to the metadata entry of a node we will end up overwriting any existing children if they share the same name, which never happens in practice since you cannot multiple children nodes/files with the same name under a single parent node/directory. This means that our metadata/nodes bucket will not be fully populated, since a parent can really only have 1 child. 

To avoid this, we use rand so we can get an actual pseudo random string. We still use a fixed length of 10 for the filename, to ensure their isn't any variance in bbolt write performance between benchmark runs. (bbolt doesn't care about the content of a KV pair since they are just interpreted as byte slices; the length, however, does matter since it controls how nodes/pages are split before writing to disk).